internal/tools/aws/cluster_management.go

```diff
@@ -57,7 +57,7 @@ func getClusterResourcesForVPC(vpcID string) (ClusterResources, error) {
 		Values: []*string{aws.String("private")},
 	})
 
-	privateSubnets, err := GetSubnetsWithFilters(privateSubnetFilter)
+	privateSubnets, err := a.GetSubnetsWithFilters(privateSubnetFilter, logger)
```

Comment:
Is this pattern preferable to making `logger` a member of `Client` so that it can be accessed inside of these methods instead of being passed in each time because of the ability to add context to the loggers when passing them in?

If so, should we add fields to these loggers when passing them in? Otherwise it seems like a lot of easily eliminated repetition to me.

Reply:
That seems to be a common pattern in this repo. In fact, somewhere in this PR (if not in this exactly case) I was told to pass logger in.

Reply:
Can we change `logger` to `logger.WithFields()` and then add a field to add debugging context at the call sites, to take advantage of passing the logger in each time?